File: Searching and Sorting Algorithms/codigo_cs/4Find Kth Character Position.cs
using System;
using System.Linq;

public class Test4
{
    public static void KthCharacter()
    {
        // your code goes here
        string[] input = Console.ReadLine().Split(' ');
        string s1 = input[0];
        char c1 = input[1][0];
        int k = int.Parse(input[2]);

        int count = 0;
        bool isFound = false;
        for (int i = 0; i < s1.Length; i++)
        {
            if (s1[i] == c1)
            {
                count++;
            }
            if (count == k)
            {
                Console.WriteLine(i);
                isFound = true;
                break;
            }
        }
        if (!isFound)
        {
            Console.WriteLine(-1);
        }

    }
}

File: Searching and Sorting Algorithms/4Find Kth Character Position.py
# cook your dish here
data=input().split()
s1=data[0]
c1=data[1]
k=int(data[2])

count=0
isFound=False
for i,ch in enumerate(s1):
    if ch==c1:
        count+=1
    if count==k:
        print(i)
        isFound=True
        break
if not isFound:
    print(-1)
